zerver/views/webhooks/__init__.py

```diff
@@ -0,0 +1,49 @@
+from django.conf.urls import url
+
+
+class Integration(object):
+    DEFAULT_LOGO_STATIC_PATH = 'static/images/integrations/logo/{name}'
+    DEFAULT_FUNCTION_PATH = 'zerver.views.webhooks.{name}.api_{name}_webhook'
+    DEFAULT_CLIENT_NAME = 'Zulip{name}Webhook'
+    DEFAULT_URL = 'api/v1/external/{name}'
+
+    def __init__(self, name, logo=None, function=None, client_name=None, url=None):
+        self.name = name
+        self.logo = logo if logo else self.DEFAULT_LOGO_STATIC_PATH.format(name=name)
+        self.function = function if function else self.DEFAULT_FUNCTION_PATH.format(name=name)
+        self.client_name = client_name if client_name else self.DEFAULT_CLIENT_NAME.format(name=name.title())
+        self.url = url if url else self.DEFAULT_URL.format(name=name)
+
+    @property
+    def url_object(self):
+        return url(self.url, self.function)
+
+
+LIST_OF_INTEGRATIONS = [
```

Comment:
Would probably use as the primary data structure something like `INTEGRATIONS` that's a dict, mapping `name` to the Integration object.  Could probably do this by renaming this to `webhook_integrations` and writing
```
INTEGRATIONS = {} # type: Dict[str, Integration]
for integration in webhook_integrations:
    INTEGRATIONS[integration.name] = integration
```

(and then in urls.py we'd use something like `for integration in INTEGRATIONS.values()`)